src/fondant/compiler.py

```diff
@@ -27,6 +27,10 @@ class Compiler(ABC):
     def compile(self, *args, **kwargs) -> None:
         """Abstract method to invoke compilation."""
 
+    @abstractmethod
+    def set_configuration(self, *args, **kwargs) -> None:
```

Comment:
Any reason to make this a public method?

Reply:
Not needed indeed, will revert it